keras_rs/src/layers/embedding/embed_reduce.py

```diff
@@ -10,6 +10,77 @@
 SUPPORTED_COMBINERS = ("mean", "sum", "sqrtn")
 
 
+def _is_supported_sparse(x: types.Tensor) -> bool:
+    """Determines if the input is a supported sparse tensor.
+
+    NOTE: Currently only works for the TensorFlow and JAX backends.
+
+    Args:
+      x: Input tensor to check for sparsity.
+
+    Returns:
+      True if `x` is a supported sparse tensor.
+    """
+    if keras.backend.backend() == "tensorflow":
+        import tensorflow as tf
+
+        return isinstance(x, tf.SparseTensor)
+    elif keras.backend.backend() == "jax":
+        from jax.experimental import sparse as jax_sparse
+
+        return isinstance(x, jax_sparse.BCOO) or isinstance(x, jax_sparse.BCSR)
+
+    return False
+
+
+def _sparse_ones_like(
+    x: types.Tensor, dtype: Optional[types.DType] = None
+) -> types.Tensor:
+    """Creates a tensor of ones with the same sparsity as the input.
+
+    This differs from `keras.ops.ones_like`, which would create a dense
+    tensor of ones.
+
+    Args:
+        x: Input sparse tensor.
+        dtype: Optional dtype for the output tensor values.
+
+    Returns:
+        Sparse tensor of ones.
+
+    Raises:
+        ValueError for unsupported sparse input type and backend.
+    """
+    dtype = dtype or x.dtype
+    if keras.backend.backend() == "tensorflow":
+        import tensorflow as tf
+
+        return tf.sparse.map_values(tf.ones_like, x, dtype=dtype)
```

Comment:
Unfortunately, `tf.sparse.map_values` loses the static shape, which almost always causes issues in graph mode. So I created this as a workaround:
https://github.com/keras-team/keras/blob/master/keras/src/backend/tensorflow/sparse.py#L25
```
sparse_with_values(x, tf.ones_like(x.values, dtype=dtype))
```

Reply:
Interesting.  That sounds like a TF bug, but maybe it's because the `map_values` function may involve multiple sparse tensors?  Anyway, I'll copy your approach until those sparse helper functions are exposed in the public API.

Reply:
Yes it is, basically all TF sparse ops have this bug. I fixed one here: https://github.com/tensorflow/tensorflow/commit/8056627d6a794f9acaff1a8165cc247bd252db31

Then I decided to fix them in keras instead.